backend/data/src/main/kotlin/io/tolgee/service/security/UserAccountService.kt

```diff
@@ -92,42 +91,22 @@ class UserAccountService(
   }
 
   @CacheEvict(cacheNames = [Caches.USER_ACCOUNTS], key = "#result.id")
+  @Transactional
   fun createUser(userAccount: UserAccount): UserAccount {
     userAccountRepository.saveAndFlush(userAccount)
     applicationEventPublisher.publishEvent(OnUserCreated(this, userAccount))
     applicationEventPublisher.publishEvent(OnUserCountChanged(this))
     return userAccount
   }
 
+  @CacheEvict(cacheNames = [Caches.USER_ACCOUNTS], key = "#result.id")
```

Comment:
Any valid turn of events would make this path a no-op. Unsure it is necessary to include.

I'm also unsure if the entity will automatically be mutated to have the correct ID; I recall the JPA spec specifying the `save` method and others may return a completely new object which should be used for further operations on the entity. *Even if it does mutate the original object, it is not something mandated by the spec and subject to one day explode out of nowhere*

Reply:
Hmm, do we actually need to evict the cache there? We can remove it from the creation methods, right?